packages/web3-types/src/apis/eth_execution_api.ts

```diff
@@ -104,10 +104,9 @@ export type TransactionUnsignedAPI =
 	| TransactionLegacyUnsignedAPI;
 
 // https://github.com/ethereum/execution-apis/blob/main/src/schemas/transaction.yaml#L211
-export type TransactionSignedAPI =
-	| Transaction1559SignedAPI
-	| Transaction2930SignedAPI
-	| TransactionLegacySignedAPI;
+export type TransactionSignedAPI = Transaction1559SignedAPI &
+	Transaction2930SignedAPI &
+	TransactionLegacySignedAPI;
```

Comment:
According to EL Specs: 
https://ethereum.github.io/execution-apis/api-documentation/ 

`gasPrice` is missing on `Transaction1559SignedAPI`, so only add `gasPrice` in it. 
Result of `getTransactionbyHash` should be `oneOf` 4844 or 2930 or 1559 or legacy tx and not all so it shouldnt be Intersection Type.

<img width="1197" alt="image" src="https://github.com/web3/web3.js/assets/86780488/6b4a4e7d-023f-4e24-8927-d1c3f392dc43">

Reply:
"openrpc": "1.2.4" json generated with [EL Specs](https://github.com/ethereum/execution-apis/tree/057892b256c181232ca22499d1801b9637192698)  have:

```
{
    "name": "eth_getTransactionByHash",
    "summary": "Returns the information about a transaction requested by transaction hash.",
    "params": [
        {
            "name": "Transaction hash",
            "required": true,
            "schema": {
                "title": "32 byte hex value",
                "type": "string",
                "pattern": "^0x[0-9a-f]{64}$"
            }
        }
    ],
    "result": {
        "name": "Transaction information",
        "schema": {
            "oneOf": [
                {
                    "title": "Not Found (null)",
                    "type": "null"
                },
                {
                    "type": "object",
                    "title": "Transaction information",
                    "required": [
                        "blockHash",
                        "blockNumber",
                        "from",
                        "hash",
                        "transactionIndex"
                    ],
                    "unevaluatedProperties": false,
                    "oneOf": [
                        {
                            "title": "Signed 4844 Transaction",
                            "type": "object",
                            "required": [
                                "accessList",
                                "blobVersionedHashes",
                                "chainId",
                                "gas",
                                "input",
                                "maxFeePerBlobGas",
                                "maxFeePerGas",
                                "maxPriorityFeePerGas",
                                "nonce",
                                "r",
                                "s",
                                "to",
                                "type",
                                "value",
                                "yParity"
                            ],
                            "properties": {
                                "type": {
                                    "title": "type",
                                    "type": "string",
                                    "pattern": "^0x([0-9,a-f,A-F]?){1,2}$"
                                },
                                "nonce": {
                                    "title": "nonce",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "to": {
                                    "title": "to address",
                                    "type": "string",
                                    "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                },
                                "gas": {
                                    "title": "gas limit",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "value": {
                                    "title": "value",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "input": {
                                    "title": "input data",
                                    "type": "string",
                                    "pattern": "^0x[0-9a-f]*$"
                                },
                                "maxPriorityFeePerGas": {
                                    "title": "max priority fee per gas",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "Maximum fee per gas the sender is willing to pay to miners in wei"
                                },
                                "maxFeePerGas": {
                                    "title": "max fee per gas",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The maximum total fee per gas the sender is willing to pay (includes the network / base fee and miner / priority fee) in wei"
                                },
                                "maxFeePerBlobGas": {
                                    "title": "max fee per blob gas",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The maximum total fee per gas the sender is willing to pay for blob gas in wei"
                                },
                                "accessList": {
                                    "title": "accessList",
                                    "type": "array",
                                    "description": "EIP-2930 access list",
                                    "items": {
                                        "title": "Access list entry",
                                        "type": "object",
                                        "additionalProperties": false,
                                        "properties": {
                                            "address": {
                                                "title": "hex encoded address",
                                                "type": "string",
                                                "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                            },
                                            "storageKeys": {
                                                "type": "array",
                                                "items": {
                                                    "title": "32 byte hex value",
                                                    "type": "string",
                                                    "pattern": "^0x[0-9a-f]{64}$"
                                                }
                                            }
                                        }
                                    }
                                },
                                "blobVersionedHashes": {
                                    "title": "blobVersionedHashes",
                                    "description": "List of versioned blob hashes associated with the transaction's EIP-4844 data blobs.",
                                    "type": "array",
                                    "items": {
                                        "title": "32 byte hex value",
                                        "type": "string",
                                        "pattern": "^0x[0-9a-f]{64}$"
                                    }
                                },
                                "chainId": {
                                    "title": "chainId",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "Chain ID that this transaction is valid on."
                                },
                                "yParity": {
                                    "title": "yParity",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The parity (0 for even, 1 for odd) of the y-value of the secp256k1 signature."
                                },
                                "r": {
                                    "title": "r",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "s": {
                                    "title": "s",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                }
                            }
                        },
                        {
                            "title": "Signed 1559 Transaction",
                            "type": "object",
                            "required": [
                                "accessList",
                                "chainId",
                                "gas",
                                "gasPrice",
                                "input",
                                "maxFeePerGas",
                                "maxPriorityFeePerGas",
                                "nonce",
                                "r",
                                "s",
                                "type",
                                "value",
                                "yParity"
                            ],
                            "properties": {
                                "type": {
                                    "title": "type",
                                    "type": "string",
                                    "pattern": "^0x2$"
                                },
                                "nonce": {
                                    "title": "nonce",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "to": {
                                    "title": "to address",
                                    "oneOf": [
                                        {
                                            "title": "Contract Creation (null)",
                                            "type": "null"
                                        },
                                        {
                                            "title": "Address",
                                            "type": "string",
                                            "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                        }
                                    ]
                                },
                                "gas": {
                                    "title": "gas limit",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "value": {
                                    "title": "value",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "input": {
                                    "title": "input data",
                                    "type": "string",
                                    "pattern": "^0x[0-9a-f]*$"
                                },
                                "maxPriorityFeePerGas": {
                                    "title": "max priority fee per gas",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "Maximum fee per gas the sender is willing to pay to miners in wei"
                                },
                                "maxFeePerGas": {
                                    "title": "max fee per gas",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The maximum total fee per gas the sender is willing to pay (includes the network / base fee and miner / priority fee) in wei"
                                },
                                "gasPrice": {
                                    "title": "gas price",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The effective gas price paid by the sender in wei. For transactions not yet included in a block, this value should be set equal to the max fee per gas. This field is DEPRECATED, please transition to using effectiveGasPrice in the receipt object going forward."
                                },
                                "accessList": {
                                    "title": "accessList",
                                    "type": "array",
                                    "description": "EIP-2930 access list",
                                    "items": {
                                        "title": "Access list entry",
                                        "type": "object",
                                        "additionalProperties": false,
                                        "properties": {
                                            "address": {
                                                "title": "hex encoded address",
                                                "type": "string",
                                                "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                            },
                                            "storageKeys": {
                                                "type": "array",
                                                "items": {
                                                    "title": "32 byte hex value",
                                                    "type": "string",
                                                    "pattern": "^0x[0-9a-f]{64}$"
                                                }
                                            }
                                        }
                                    }
                                },
                                "chainId": {
                                    "title": "chainId",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "Chain ID that this transaction is valid on."
                                },
                                "yParity": {
                                    "title": "yParity",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The parity (0 for even, 1 for odd) of the y-value of the secp256k1 signature."
                                },
                                "v": {
                                    "title": "v",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "For backwards compatibility, `v` is optionally provided as an alternative to `yParity`. This field is DEPRECATED and all use of it should migrate to `yParity`."
                                },
                                "r": {
                                    "title": "r",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "s": {
                                    "title": "s",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                }
                            }
                        },
                        {
                            "title": "Signed 2930 Transaction",
                            "type": "object",
                            "required": [
                                "accessList",
                                "chainId",
                                "gas",
                                "gasPrice",
                                "input",
                                "nonce",
                                "r",
                                "s",
                                "type",
                                "value",
                                "yParity"
                            ],
                            "properties": {
                                "type": {
                                    "title": "type",
                                    "type": "string",
                                    "pattern": "^0x1$"
                                },
                                "nonce": {
                                    "title": "nonce",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "to": {
                                    "title": "to address",
                                    "oneOf": [
                                        {
                                            "title": "Contract Creation (null)",
                                            "type": "null"
                                        },
                                        {
                                            "title": "Address",
                                            "type": "string",
                                            "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                        }
                                    ]
                                },
                                "gas": {
                                    "title": "gas limit",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "value": {
                                    "title": "value",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "input": {
                                    "title": "input data",
                                    "type": "string",
                                    "pattern": "^0x[0-9a-f]*$"
                                },
                                "gasPrice": {
                                    "title": "gas price",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The gas price willing to be paid by the sender in wei"
                                },
                                "accessList": {
                                    "title": "accessList",
                                    "type": "array",
                                    "description": "EIP-2930 access list",
                                    "items": {
                                        "title": "Access list entry",
                                        "type": "object",
                                        "additionalProperties": false,
                                        "properties": {
                                            "address": {
                                                "title": "hex encoded address",
                                                "type": "string",
                                                "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                            },
                                            "storageKeys": {
                                                "type": "array",
                                                "items": {
                                                    "title": "32 byte hex value",
                                                    "type": "string",
                                                    "pattern": "^0x[0-9a-f]{64}$"
                                                }
                                            }
                                        }
                                    }
                                },
                                "chainId": {
                                    "title": "chainId",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "Chain ID that this transaction is valid on."
                                },
                                "yParity": {
                                    "title": "yParity",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The parity (0 for even, 1 for odd) of the y-value of the secp256k1 signature."
                                },
                                "v": {
                                    "title": "v",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "For backwards compatibility, `v` is optionally provided as an alternative to `yParity`. This field is DEPRECATED and all use of it should migrate to `yParity`."
                                },
                                "r": {
                                    "title": "r",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "s": {
                                    "title": "s",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                }
                            }
                        },
                        {
                            "title": "Signed Legacy Transaction",
                            "type": "object",
                            "required": [
                                "gas",
                                "gasPrice",
                                "input",
                                "nonce",
                                "r",
                                "s",
                                "type",
                                "v",
                                "value"
                            ],
                            "properties": {
                                "type": {
                                    "title": "type",
                                    "type": "string",
                                    "pattern": "^0x0$"
                                },
                                "nonce": {
                                    "title": "nonce",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "to": {
                                    "title": "to address",
                                    "oneOf": [
                                        {
                                            "title": "Contract Creation (null)",
                                            "type": "null"
                                        },
                                        {
                                            "title": "Address",
                                            "type": "string",
                                            "pattern": "^0x[0-9,a-f,A-F]{40}$"
                                        }
                                    ]
                                },
                                "gas": {
                                    "title": "gas limit",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "value": {
                                    "title": "value",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "input": {
                                    "title": "input data",
                                    "type": "string",
                                    "pattern": "^0x[0-9a-f]*$"
                                },
                                "gasPrice": {
                                    "title": "gas price",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "The gas price willing to be paid by the sender in wei"
                                },
                                "chainId": {
                                    "title": "chainId",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$",
                                    "description": "Chain ID that this transaction is valid on."
                                },
                                "v": {
                                    "title": "v",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "r": {
                                    "title": "r",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                },
                                "s": {
                                    "title": "s",
                                    "type": "string",
                                    "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                                }
                            }
                        }
                    ],
                    "properties": {
                        "blockHash": {
                            "title": "block hash",
                            "type": "string",
                            "pattern": "^0x[0-9a-f]{64}$"
                        },
                        "blockNumber": {
                            "title": "block number",
                            "type": "string",
                            "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                        },
                        "from": {
                            "title": "from address",
                            "type": "string",
                            "pattern": "^0x[0-9,a-f,A-F]{40}$"
                        },
                        "hash": {
                            "title": "transaction hash",
                            "type": "string",
                            "pattern": "^0x[0-9a-f]{64}$"
                        },
                        "transactionIndex": {
                            "title": "transaction index",
                            "type": "string",
                            "pattern": "^0x([1-9a-f]+[0-9a-f]*|0)$"
                        }
                    }
                }
            ]
        }
    }
},
```